consensus/polybft/bitmap/bitmap.go

```diff
@@ -0,0 +1,37 @@
+package bitmap
+
+// Bitmap Index 0 is LSB from the first bitmap byte
+type Bitmap []byte
+
+func (b *Bitmap) Set(idx uint64) {
+	index := idx / 8
+	*b = extendByteSlice(*b, int(index)+1)
+
+	bit := uint8(1 << (idx % 8))
+	(*b)[idx/8] |= bit
+}
+
+func (b *Bitmap) Len() uint64 {
+	return uint64(len(*b) * 8)
+}
+
+func (b *Bitmap) IsSet(idx uint64) bool {
+	if length := b.Len(); length <= idx {
+		return false
+	}
+
+	bit := uint8(1 << (idx % 8))
+
+	return (*b)[idx/8]&bit == bit
+}
+
+func extendByteSlice(b []byte, needLen int) []byte {
+	// for example if we need to store idx 277 we need 35 bytes
+	// But if have slice which length is 5 bytes we need to add additional 30 bytes
+	// append function is smart enough to use capacity of slice if needed otherwise it will create new slice
```

Comment:
Why is this comment required?
It's basic `append` functionality, no need to explicitly mention it

Reply:
hmm

<img width="251" alt="image" src="https://user-images.githubusercontent.com/47026/195370278-1dc38878-7b2a-4bcc-9970-e770b8319dc7.png">

Repeated up to 4 times, sounds like a global helper?

Reply:
Yep, I think all of those are the same.